transaction-error/src/lib.rs

```diff
@@ -163,7 +168,18 @@ impl fmt::Display for TransactionError {
              => f.write_str("This transaction has already been processed"),
             Self::BlockhashNotFound
              => f.write_str("Blockhash not found"),
-            Self::InstructionError(idx, err) =>  write!(f, "Error processing Instruction {idx}: {err}"),
+            Self::InstructionError(
+                index,
+                err,
+                _inner_instruction_index,
+                _responsible_program_account_index,
+            )
+             // NOTE: We intentionally do not augment the error message in the event that the error
+             // carries the index of the inner instruction or the account index of the responsible
+             // program. While it would add value to the log, to do so now would also break any log
+             // parser that presumed the log format to be immutable for all time (eg.
+             // https://tinyurl.com/3uuczr68).
+             =>  write!(f, "Error processing Instruction {index}: {err}"),
```

Comment:
I think that's fine for now, people can also use the `Debug` formatting to get every piece of data.

We could consider creating a new enum variant, but then the UMI parser would be totally useless, which is even worse in my opinion.